nucypher/blockchain/eth/actors.py

```diff
@@ -1194,15 +1200,69 @@ def __init__(self,
 
         # Workers cannot be started without being assigned a stake first.
         if is_me:
+            if block_until_ready:
+                self.block_until_ready()
             self.stakes = StakeList(registry=self.registry, checksum_address=self.checksum_address)
             self.stakes.refresh()
-            if check_active_worker and not len(self.stakes):
-                raise self.DetachedWorker(f"{self.__worker_address} is not bonded to {self.checksum_address}.")
 
             self.work_tracker = work_tracker or WorkTracker(worker=self)
             if start_working_now:
                 self.work_tracker.start(act_now=False)
 
+    def block_until_ready(self, poll_rate: int = None, timeout: int = None):
+        """
+        Polls the staking_agent and blocks until the staking address is not
+        a null address for the given worker_address. Once the worker is bonded, it returns the staker address.
+        """
+        if not self.__worker_address:
+            raise RuntimeError("No worker address available")
+
+        timeout = timeout or self.BONDING_TIMEOUT
+        poll_rate = poll_rate or self.BONDING_POLL_RATE
+        staking_agent = ContractAgency.get_agent(StakingEscrowAgent, registry=self.registry)
+        client = staking_agent.blockchain.client
+        start = maya.now()
+
+        emitter = StdoutEmitter()  # TODO: Make injectable, or embed this logic into Ursula
+        emitter.message("Waiting for bonding and funding...")
+
+        counter, funded, bonded = 0, False, False
```

Comment:
Why do we have a counter here? Seems like it's not being used.

Reply:
:heavy_check_mark: